### PR TITLE
Feature/dependency refactoring

### PR DESCRIPTION
**Pull request by kptfh**

### CODE REVIEW

**Comment by vasilievip** [comment]
why you are trying to merge into master and not develop?

File: feign-reactor-core/src/test/java/reactivefeign/StatusHandlerTest.java [hunk 1]
```diff
@@ -129,6 +129,7 @@ public void shouldThrowOnStatusCode() {
             throwOnStatus(
                 status -> status == HttpStatus.SC_SERVICE_UNAVAILABLE,
                 (methodTag, response) -> new RetryableException(
+                        //response.status(),
```
Comment (vasilievip) [inline feign-reactor-core/src/test/java/reactivefeign/StatusHandlerTest.java]: why is this commented?

File: feign-reactor-jetty/src/test/java/reactivefeign/jetty/h2c/AllFeaturesTest.java [hunk 2]
```diff
@@ -16,57 +16,48 @@
 
 package reactivefeign.jetty.h2c;
 
-import org.eclipse.jetty.http2.server.HTTP2CServerConnectionFactory;
-import org.eclipse.jetty.server.HttpConfiguration;
-import org.eclipse.jetty.server.ServerConnector;
+import org.junit.Ignore;
```
Comment (vasilievip) [inline feign-reactor-jetty/src/test/java/reactivefeign/jetty/h2c/AllFeaturesTest.java]: why do you need @Ignore?

File: feign-reactor-rx2/src/test/java/reactivefeign/rx2/StatusHandlerTest.java [hunk 3]
```diff
@@ -53,7 +53,9 @@ public void shouldThrowRetryException() throws InterruptedException {
     IcecreamServiceApi client = builder()
         .statusHandler(throwOnStatus(
             status -> status == HttpStatus.SC_SERVICE_UNAVAILABLE,
-            (methodTag, response) -> new RetryableException("Should retry on next node",
+            (methodTag, response) -> new RetryableException(
+                    //response.status(),
```
Comment (vasilievip) [inline feign-reactor-rx2/src/test/java/reactivefeign/rx2/StatusHandlerTest.java]: why this commented?

File: feign-reactor-test/feign-reactor-server-configurations/src/test/java/reactivefeign/wiremock/WireMockServerConfigurations.java [hunk 4]
```diff
@@ -25,19 +30,22 @@ protected ServerConnector createHttpConnector(
...
+                                http2CFactory.setMaxConcurrentStreams(maxConcurrentCalls);
 
                                 return createServerConnector(
                                         bindAddress,
                                         jettySettings,
                                         port,
                                         listener,
-                                        new HttpConnectionFactory(httpConfig),
-                                        http2CFactory
+                                        includeH1
```
Comment (vasilievip) [inline feign-reactor-test/feign-reactor-server-configurations/src/test/java/reactivefeign/wiremock/WireMockServerConfigurations.java]: why not always include http1  and drop boolean input parameter?

File: pom.xml [hunk 5]
```diff
@@ -253,306 +161,40 @@
...
-            <!--Jetty part-->
-            <dependency>
-                <groupId>org.eclipse.jetty</groupId>
-                <artifactId>jetty-reactive-httpclient</artifactId>
-                <version>${jetty-reactive-httpclient.version}</version>
-            </dependency>
-
+            <!-- Other -->
             <dependency>
                 <groupId>io.kptfh.reactivejson</groupId>
```
Comment (vasilievip) [inline pom.xml]: io.kptfh.reactivejson is this still required?

File: feign-reactor-jetty/src/test/java/reactivefeign/jetty/h2c/AllFeaturesTest.java (hunk 2 above)
Comment (kptfh) [reply feign-reactor-jetty/src/test/java/reactivefeign/jetty/h2c/AllFeaturesTest.java]: fixed

File: feign-reactor-rx2/src/test/java/reactivefeign/rx2/StatusHandlerTest.java (hunk 3 above)
Comment (kptfh) [reply feign-reactor-rx2/src/test/java/reactivefeign/rx2/StatusHandlerTest.java]: fixed

File: feign-reactor-test/feign-reactor-server-configurations/src/test/java/reactivefeign/wiremock/WireMockServerConfigurations.java (hunk 4 above)
Comment (kptfh) [reply feign-reactor-test/feign-reactor-server-configurations/src/test/java/reactivefeign/wiremock/WireMockServerConfigurations.java]: jetty may work directly with h2c
java11 needs http1 to make upgrade

File: pom.xml (hunk 5 above)
Comment (kptfh) [reply pom.xml]: yes

**Pull request merged**